src/sonic-host-services/tests/hostcfgd/hostcfgd_passwh_test.py

```diff
@@ -0,0 +1,200 @@
+        max_days = -1
+        REGEX_MAX_DAYS = r'^PASS_MAX_DAYS[ \t]*(?P<max_days>\d*)'
+
+        with open(login_file, 'r') as f:
+            login_def_data = f.readlines()
+
+        for line in login_def_data:
+            m1 = re.match(REGEX_MAX_DAYS, line)
+            if m1:
+                max_days = int(m1.group("max_days"))
+                break
+        return max_days
+
+    def get_passw_warn_days(self, login_file):
+        warn_days = -1
+        REGEX_WARN_DAYS = r'^PASS_WARN_AGE[ \t]*(?P<warn_days>\d*)'
+
+        with open(login_file, 'r') as f:
+            login_def_data = f.readlines()
+
+        for line in login_def_data:
+            m1 = re.match(REGEX_WARN_DAYS, line)
+            if m1:
+                warn_days = int(m1.group("warn_days"))
+                break
+        return warn_days
+
+    """
+        Check different config
+    """
+    def check_config(self, test_name, test_data, config_name):
+        t_path = templates_path
+        op_path = output_path + "/" + test_name + "_" + config_name
+        sop_path = sample_output_path + "/" +  test_name + "_" + config_name
+
+        hostcfgd.PAM_PASSWORD_CONF_TEMPLATE = t_path + "/common-password.j2"
+        hostcfgd.PAM_AUTH_CONF_TEMPLATE = t_path + "/common-auth-sonic.j2"
+        hostcfgd.NSS_TACPLUS_CONF_TEMPLATE = t_path + "/tacplus_nss.conf.j2"
+        hostcfgd.NSS_RADIUS_CONF_TEMPLATE = t_path + "/radius_nss.conf.j2"
+        hostcfgd.PAM_RADIUS_AUTH_CONF_TEMPLATE = t_path + "/pam_radius_auth.conf.j2"
+        hostcfgd.PAM_PASSWORD_CONF = op_path + "/common-password"
+        hostcfgd.ETC_LOGIN_DEF = op_path + "/login.defs"
+        hostcfgd.PAM_AUTH_CONF = op_path + "/common-auth-sonic"
+        hostcfgd.NSS_TACPLUS_CONF = op_path + "/tacplus_nss.conf"
+        hostcfgd.NSS_RADIUS_CONF = op_path + "/radius_nss.conf"
+        hostcfgd.NSS_CONF = op_path + "/nsswitch.conf"
+        hostcfgd.ETC_PAMD_SSHD = op_path + "/sshd"
+        hostcfgd.ETC_PAMD_LOGIN = op_path + "/login"
+        hostcfgd.RADIUS_PAM_AUTH_CONF_DIR = op_path + "/"
+
+        shutil.rmtree(op_path, ignore_errors=True)
+        os.mkdir(op_path)
+
+
+        shutil.copyfile(sop_path + "/login.defs.old", op_path + "/login.defs")
+        MockConfigDb.set_config_db(test_data[config_name])
+        host_config_daemon = hostcfgd.HostConfigDaemon()
+
+        aaa = host_config_daemon.config_db.get_table('AAA')
+
+        try:
+            passwh_table = host_config_daemon.config_db.get_table('PASSW_HARDENING')
+        except:
+            # TODO: raise an error?
+            passwh_table = []
+
+        host_config_daemon.aaacfg.load(aaa,[],[],[],[],passwh_table)
+
+        # dcmp = filecmp.dircmp(sop_path, op_path)
+        # print(hostcfgd.PAM_PASSWORD_CONF)
+        # /sonic/src/sonic-host-services/tests/hostcfgd/output/PASSWORD_HARDENING_default_values/common-password
+        diff_output = ""
+        # name = 'common-password'
+
+        files_to_compare = ['common-password']
```

Comment:
I believe only common-password and login.defs files are used. But there are so many files defined like (login.old, login.defs.old, sshd, common-auth-sonic, sshd, radius and tacplus files,etc.
Why are these files needed in this PR? If they are not needed please remove and keep the bare minimum to run the mock tests.

Reply:
Done. (my unitest only use 3 files, so I removed the others)